de.tu_bs.cs.isf.e4cf.compare.data_structures_editor/src/de/tu_bs/cs/isf/e4cf/compare/data_structures_editor/utilities/TreeViewUtilities.java

```diff
@@ -0,0 +1,274 @@
+			if (treeView.getSelectionModel().getSelectedIndices().size() == 1) {
+				switchToPart(DataStructuresEditorST.PROPERTIES_VIEW_ID, services);
+				services.eventBroker.send("nodePropertiesEvent",
+						treeView.getSelectionModel().getSelectedItem().getValue());
+			}
+		});
+
+		return treeView;
+	}
+
+	/**
+	 * 
+	 * @param item
+	 * @param name
+	 * @return
+	 */
+	public static List<TreeItem<NodeUsage>> searchTreeItem(TreeItem<NodeUsage> item, String name) {
+		if (item.getValue().toString().contains(name)) {
+			searchList.add(item);
+		}
+		List<TreeItem<NodeUsage>> result = new ArrayList<TreeItem<NodeUsage>>();
+		for (TreeItem<NodeUsage> child : item.getChildren()) {
+			result.addAll(searchTreeItem(child, name));
+			if (result.size() < 1) {
+				searchList.addAll(result);
+			}
+		}
+		return searchList;
+	}
+
+	public static List<TreeItem<NodeUsage>> getSearchList() {
+		return searchList;
+	}
+
+	public static void clearSearchList() {
+		searchList.clear();
+	}
+
+	public static void serializesTree(TreeView<NodeUsage> treeView) {
+		File file = new File(RCPContentProvider.getCurrentWorkspacePath() + "/" + treeName);
+		writeToFile(file, treeView);
+	}
+
+	public static void serializesTree(TreeView<NodeUsage> treeView, String newFileName) {
+		File file = new File(RCPContentProvider.getCurrentWorkspacePath() + "/" + newFileName);
+		writeToFile(file, treeView);
+	}
+
+	public static void extractTree(TreeView<NodeUsage> treeView, String newFileName, List<TreeItem> tempList) {
+		File file = new File(RCPContentProvider.getCurrentWorkspacePath() + "/" + newFileName);
+		try {
+			FileWriter writer = new FileWriter(file);
+			for (TreeItem<NodeUsage> node : tempList) {
+				writer.write(node.getValue().toString());
+				writer.write("\n");
+			}
+			writer.close();
+			System.out.println("Tree: " + file.getAbsolutePath() + " stored.");
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+
+	public static String getInput(String displayedDialog) {
+		TextInputDialog td = new TextInputDialog();
+		td.setHeaderText(displayedDialog);
+		td.setGraphic(null);
+		td.setTitle("Dialog");
+		td.showAndWait();
+		String s = td.getEditor().getText();
+		if (s.equals("") || s.equals(null)) {
+			throw new NullPointerException();
+		}
+		return s;
+
+	}
+
+	public static TreeItem<NodeUsage> getCurrentSearchItem(List<TreeItem<NodeUsage>> resultList) {
+		TreeItem<NodeUsage> currentItem = new TreeItem<NodeUsage>();
+		if (getSearchCounter() < resultList.size()) {
+			currentItem = resultList.get(getSearchCounter());
+			incrementSearchCounter();
+		} else {
+			setSearchCounter(0);
+			currentItem = resultList.get(getSearchCounter());
+			incrementSearchCounter();
+		}
+		return currentItem;
+	}
+
+	public static int getSearchCounter() {
+		return searchCounter;
+	}
+
+	public static void setSearchCounter(int i) {
+		searchCounter = i;
+	}
+
+	public static void incrementSearchCounter() {
+		searchCounter++;
+	}
+
```

Comment:
Ich glaube das kann man auch auslagern in Utilitie Klassen bzw. ich glaube das existiert auch schon